### PR TITLE
Interfaces - Handle multiple fragments.

### PR DESCRIPTION
**Pull request by Pruxis**
In this PR I have changed how the fragments are handled a bit.

Instead of relying on the node matching I would match all nodes that are related with the relation you are fetching. And pass the label as a FRAGMENT_TYPE so the default resolver can do the rest.

This will support default interface queries & multiple fragments and will still return the correct type.

However if your graph would contain multiple labels for a node we would have to pass the entire array of labels and check that instead in the default resolver, I didn't do that yet but if core contributors would like to see this "improvement" I can surely add this here.

I'm quite blocked by this PR so it would be great if there is some feedback here that we can solve this issue as fast as possible.

Looking forward to contribute more to this package as we started to rely heavily on neo4j & apollo.

### CODE REVIEW

**Comment by Pruxis** [comment]
As I was testing my current implementation I noticed that attempting to query subfields that have @relation / @cypher or just a type that isn't one of the default scalars they do not seem to get resolved.

I'll have to dig deeper to find out why..

**Comment by Pruxis** [comment]
Eureka! @johnymontana I've also solved an issue when you query fields of a type that do not exist on the interface that the type implements. I've done this by passing the schemaType of the type of the fragment instead of the interface schemaType.

Example:

interface Person {
  id: ID
}

type Man {
  id: ID
  beard: Boolean
}

then beard was not queryable before this fix.

**Comment by Pruxis** [comment]
There are still several issues regarding to interfaces that have not been handled yet such as:
- Fragment spreading in an interface
- Fragment spreading in a type that implements an interface.

Currently just having working fragments is sufficient for my use case (so far) so I'll touch those if anyone requires them.

**Comment by johnymontana** [comment]
Thanks @Pruxis! Would you mind adding a couple of tests showing some examples of what this fixes?

We were envisioning interfaces making use of multiple labels in the database. For example:


```
interface Person {
  name: String
}

type User implements Person {
  name: String
  userId: ID
}

type Employee implements Person {
  name: String
  employeeId: ID
}
```

would correspond in nodes in the database `(:Person:User)` and `(:Person:Employee)`. In that case, we would need some logic to figure out which label is the interface and which is the implementing type for the `FRAGMENT_TYPE` value. Of course, we would need to add this logic to the generated Create mutations to add both labels, which we haven't don't yet, so I'm OK with not handling the case of multiple labels for now.

File: src/translate.js [hunk 1]
```diff
@@ -688,13 +690,11 @@ const customQuery = ({
 }) => {
   const safeVariableName = safeVar(variableName);
   const [subQuery, subParams] = buildCypherSelection({
-    initial: '',
     cypherParams,
     selections,
     variableName,
     schemaType,
-    resolveInfo,
-    paramIndex: 1
```
Comment (johnymontana) [inline src/translate.js]: @Pruxis Is there a reason for removing `paramIndex` here?

Comment (Pruxis) [reply src/translate.js]: I think paramIndex defaults to 1, I'm not sure, it's been a month since I did this change

Comment (johnymontana) [reply src/translate.js]: Yup, looks like paramIndex default is 1. Thanks

**Comment by kraney** [comment]
I'd just like to chime in that I do in fact use multiple labels and need this fix to support them. I really appreciate the work that has been done so far.

Also, not every node pointed to with the given relationship type would be candidates for the field. Some can be a completely different type that doesn't implement the interface. These are available on a different field referencing a different interface, but using the same relationship type. Does the default resolver handle this case?

E.g.
```
interface A {
}

type B implements A {
}

type C implements A {
}

interface X {
}

type Y implements X {
}

type Z implements X {
}

type Container {
  exes: [X!]! @relation(name: "CONTAINS", direction: "OUT")
  ays: [A!]! @relation(name: "CONTAINS", direction: "OUT")
}
```
...and where the nodes are labeled :A:B, :A:C, :X:Y, or :X:Z. Possibly with other labels as well.

**Pull request merged**